sets/set.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sets
+
+import "sort"
+
+type Ordered interface {
```

Comment:
"golang.org/x/exp/constraints" has Ordered - should we use that? or at least define this the same way (everything is `~type`)?

Reply:
If we need go 1.20, we need to delay adding this until the oldest supported release is on 1.20 - is that worthwhile?

Would it be a compatible change to move from `Ordered` to `comparable` later?  Obviously all ordered types are comparable...

Reply:
alternately: `sets.Set[any]` seems like a terrible idea ANYWAY, so do we care if it doesn't work on older versions (corner-case of a corner-case)?